pkg/envoy/lds/inmesh.go

```diff
@@ -37,14 +38,14 @@ func (lb *listenerBuilder) getInboundMeshFilterChains(proxyService service.MeshS
 	switch strings.ToLower(proxyService.Protocol) {
 	case constants.ProtocolHTTP, constants.ProtocolGRPC:
 		// Filter chain for HTTP port
-		filterChainForPort, err := lb.getInboundMeshHTTPFilterChain(proxyService, uint32(proxyService.TargetPort))
+		filterChainForPort, err := lb.getInboundMeshHTTPFilterChain(proxyService, uint32(proxyService.TargetPort), lb.cfg.GetMeshConfig().Spec.Sidecar)
```

Comment:
Please store the `SidecarSpec` as a part of the `listenerBuilder` struct so you don't have to pass this around in every function call.

Reply:
Actually since `listenerBuilder` contains the configurator, `SidecarSpec` is stored in `listenerBuilding` already. We only can minimize the argument passing.

Reply:
For now, you can do either of the following:
1. Cache SidecarSpec in listenerBuilder
2. Cache MeshConfig in listenerBuilder
3. Use `lb.cfg.Xyz()` in lds/*

We want to avoid unnecessary access to the informer caches, as these incur a perf issue due to RW locks. So it's best to cache these in listenerBuilder for this purpose instead of option 3 above. The existing usage of 3 can be refactored later on.

Reply:
Ok I see the intentions behind now.